appcelerator/titanium_mobile
# feat: add "uprightWidth" and "uprightHeight" properties to Ti.Blob

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28093

**Summary:**
- These new properties provide image width/height when factoring in JPEG EXIF orientation.
- Needed since Android does not auto-rotate the image when loading into memory. Will only auto-rotate when displaying on-screen. Meaning "width" and "height" represent the image's unrotated dimensions.
- In the future, we should refactor our Android image loading code to load it upright like iOS, but this would be a large undertaking that is best done in a major version of Titanium

**Note:**
If we're not comfortable with adding these properties to the SDK, then alternatively we can add a method to our "ti.imagefactory" to fetch an image's upright dimensions instead.

**Test:**
1. Build and run `DownloadImageBlobTest.js` attached to [TIMOB-28093](https://jira.appcelerator.org/browse/TIMOB-28093) on Android.
2. Make sure device has Internet access.
3. Tap on the "Download" button.
4. Select "JPEG - EXIF Rotate 90" in dialog.
5. Verify image is upright and scaled proportionally (not stretched) to fit the window.
6. Repeat steps 3-5 for "JPEG - EXIF Rotate 180".
7. Repeat steps 3-5 for "JPEG - EXIF Rotate 270".
8. Repeat steps 3-5 for "JPEG - EXIF Flip Horizontal".
9. Repeat steps 3-5 for "JPEG - EXIF Flip Vertical".
10. Repeat steps 3-5 for "JPEG - EXIF Transpose".
11. Repeat steps 3-5 for "JPEG - EXIF Transverse".
12. Repeat all of the above on iOS.

_(Note that the EXIF Flip, Transpose, and Transverse tests will appear as mirrored. Fixing the mirroring effect would require a larger refactoring of our code in the future.)_
